packages/core/tests/controllers/SwapController.test.ts

```diff
@@ -1,78 +1,72 @@
-import { beforeAll, describe, expect, it } from 'vitest'
+import { beforeAll, beforeEach, describe, expect, it } from 'vitest'
 import { AccountController, SwapController, type SwapTokenWithBalance } from '../../index.js'
 import { prices, tokenInfo } from '../mocks/SwapController.js'
 import { INITIAL_GAS_LIMIT } from '../../src/controllers/SwapController.js'
+import { SwapCalculationUtil } from '../../src/utils/SwapCalculationUtil.js'
 
 // - Mocks ---------------------------------------------------------------------
 const caipAddress = 'eip155:1:0x123'
 const gasLimit = BigInt(INITIAL_GAS_LIMIT)
 const gasFee = BigInt(455966887160)
 
-const sourceTokenAddress = 'eip155:137:0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+const networkTokenAddress = 'eip155:137:0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
 
-const sourceToken = tokenInfo[0] as SwapTokenWithBalance
+// MATIC
+const networkToken = tokenInfo[0] as SwapTokenWithBalance
+// FOX
 const toToken = tokenInfo[1] as SwapTokenWithBalance
 
 // - Helpers
-function setSourceTokenAmount(value: string) {
-  SwapController.setSourceTokenAmount(value)
-  const toTokenAmount = SwapController.getToAmount()
-  const toTokenValues = SwapController.getToTokenValues(toTokenAmount, toToken?.decimals)
-  SwapController.state.toTokenAmount = toTokenValues.toTokenAmount
-  SwapController.state.toTokenPriceInUSD = toTokenValues.toTokenPriceInUSD
+function initializeSwapState(_sourceToken: SwapTokenWithBalance, _toToken: SwapTokenWithBalance) {
+  SwapController.state.tokensPriceMap = prices
+  SwapController.state.networkPrice = prices[networkTokenAddress].toString()
+  SwapController.state.networkBalanceInUSD = networkToken.quantity.numeric
+  SwapController.state.gasPriceInUSD = SwapCalculationUtil.getGasPriceInUSD(
+    SwapController.state.networkPrice,
+    gasLimit,
+    gasFee
+  )
+  SwapController.setSourceToken(_sourceToken)
+  SwapController.state.sourceTokenPriceInUSD = _sourceToken.price
+  SwapController.setToToken(_toToken)
+  SwapController.state.toTokenPriceInUSD = _toToken.price
+  SwapController.state.toTokenAmount = SwapCalculationUtil.getToTokenAmount({
+    sourceToken: _sourceToken,
+    sourceTokenAmount: '1',
+    sourceTokenPrice: _sourceToken.price,
+    toToken: _toToken,
+    toTokenPrice: _toToken.price
+  })
```

Comment:
no way to actually set this using regular `SwapController` flows?

Reply:
Are you asking for setting the `toTokenAmount ` value or all other things? These are the values when the SwapController get initialized. They are set with different endpoint responses. The only thing extra I'm mocking that the user selected the `toToken` and calculated the `toTokenAmount`. It looks a bit weird that we are using `SwapCalculationUtil` in the SwapController test but if I wrap it with a function in `SwapController` that would be unnecessary.

Reply:
I guess the proper way would be to mock the API response directly. 
There's a couple of examples in the `ApiController` tests